### PR TITLE
Add collection of basic system resources usage

### LINKED ISSUE #3: Metrics for executor process CPU and memory usage

### PR DESCRIPTION
**Pull request by medzin**
This commit adds collection of basic system resources (cpu and memory)
usage by the executor. Metrics can be sent to Graphite or periodically
printed to stderr. It resolves #3

### CODE REVIEW

File: metrics/graphite_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package metrics
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIfFailsToSetupGraphiteWithInvalidConfig(t *testing.T) {
```
Comment (AlfredBroda) [inline metrics/graphite_test.go]: Is there no way to test a positive outcome of the configuration process at least? Our sending some basic metrics to a mock server?

Comment (medzin) [reply metrics/graphite_test.go]: I can add a test case with a valid hostname, will it be ok?

Comment (medzin) [reply metrics/graphite_test.go]: @AlfredBroda test case added

File: metrics/system.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	total := t.User + t.System + t.Nice + t.Iowait + t.Irq + t.Softirq +
+		t.Steal + t.Guest + t.GuestNice + t.Stolen
+
+	return total, nil
+}
+
+// Memory returns a number of bytes used by executor process.
+func Memory() (uint64, error) {
```
Comment (janisz) [inline metrics/system.go]: We can use [`CaptureRuntimeMemStats`](https://godoc.org/github.com/rcrowley/go-metrics#CaptureRuntimeMemStats)  instead

Comment (medzin) [reply metrics/system.go]: @janisz fixed

File: metrics/system.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+// CPUTime returns an amount of CPU time (in seconds) assigned to executor process
+// by system.
+func CPUTime() (float64, error) {
+	p := getExecutorProcess()
+	t, err := p.Times()
+	if err != nil {
+		return 0, fmt.Errorf("Unable to get CPU times: %s", err)
+	}
+
+	total := t.User + t.System + t.Nice + t.Iowait + t.Irq + t.Softirq +
```
Comment (janisz) [inline metrics/system.go]: Use [`t.Total`](https://godoc.org/github.com/shirou/gopsutil/cpu#TimesStat.Total) instead

Comment (janisz) [reply metrics/system.go]: Or add comment why not using it

Comment (medzin) [reply metrics/system.go]: Commented.

**Pull request merged**